Add label for stage disk, delete the bootindex on s390x

fbq815 opened the pull request:
Virtual_blk: add label for stage disk to make sure right drive was tested while
dd testing

ID: 2119690

Signed-off-by: Boqiao Fu <bfu@redhat.com>

fbq815 commented:
Test result:
JOB ID     : f1a414cf7ab62dd572bf3743cf383fa526b06cca
JOB LOG    : /root/avocado/job-results/job-2022-08-19T04.49-f1a414c/job.log
 (01/30) repeat1.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_qcow2_format.s390-virtio: STARTED
 (01/30) repeat1.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_qcow2_format.s390-virtio: PASS (63.05 s)
 (02/30) repeat1.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_raw_format.s390-virtio: STARTED
 (02/30) repeat1.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_raw_format.s390-virtio: PASS (58.13 s)
 (03/30) repeat1.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_luks_format.s390-virtio: STARTED
 (03/30) repeat1.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_luks_format.s390-virtio: PASS (62.61 s)
 (04/30) repeat2.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_qcow2_format.s390-virtio: STARTED
 (04/30) repeat2.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_qcow2_format.s390-virtio: PASS (63.43 s)
 (05/30) repeat2.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_raw_format.s390-virtio: STARTED
 (05/30) repeat2.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_raw_format.s390-virtio: PASS (58.19 s)
 (06/30) repeat2.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_luks_format.s390-virtio: STARTED
 (06/30) repeat2.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_luks_format.s390-virtio: PASS (63.57 s)
 (07/30) repeat3.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_qcow2_format.s390-virtio: STARTED
 (07/30) repeat3.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_qcow2_format.s390-virtio: PASS (63.42 s)
(08/30) repeat3.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_raw_format.s390-virtio: STARTED
 (08/30) repeat3.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_raw_format.s390-virtio: PASS (57.46 s)
 (09/30) repeat3.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_luks_format.s390-virtio: STARTED
 (09/30) repeat3.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (62.78 s)
 (10/30) repeat4.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (10/30) repeat4.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.81 s)
 (11/30) repeat4.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (11/30) repeat4.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (58.12 s)
 (12/30) repeat4.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (12/30) repeat4.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (64.58 s)
 (13/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (13/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.47 s)
 (14/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (14/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (58.82 s)
 (15/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (15/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (63.98 s)
 (15/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_luks_format.s390-virtio: STARTED
 (15/30) repeat5.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_luks_format.s390-virtio: PASS (63.98 s)
 (16/30) repeat6.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.wi
th_qcow2_format.s390-virtio: STARTED
 (16/30) repeat6.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.00 s)
 (17/30) repeat6.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (17/30) repeat6.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (56.83 s)
 (18/30) repeat6.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (18/30) repeat6.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (62.90 s)
 (19/30) repeat7.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (19/30) repeat7.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.68 s)
 (20/30) repeat7.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (20/30) repeat7.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (58.30 s)
(21/30) repeat7.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (21/30) repeat7.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (62.11 s)
 (22/30) repeat8.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (22/30) repeat8.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.30 s)
 (23/30) repeat8.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (23/30) repeat8.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (58.92 s)
 (24/30) repeat8.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (24/30) repeat8.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (63.78 s)
 (25/30) repeat9.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (25/30) repeat9.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.49 s)
 (26/30) repeat9.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (26/30) repeat9.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (58.75 s)
 (27/30) repeat9.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (27/30) repeat9.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (64.01 s)
 (28/30) repeat10.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (28/30) repeat10.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.55 s)
 (29/30) repeat10.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (29/30) repeat10.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (58.17 s)
 (30/30) repeat10.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (30/30) repeat10.Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (62.58 s)
RESULTS    : PASS 30 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-08-19T04.49-f1a414c/results.html
JOB TIME   : 1878.94 s

@zhenyzha @maxujun please help me review if this patch works on your platform

@PaulYuuu Please help me review this patch

fbq815 commented:
@qingwangrh please help me check if this works on x86 as well, thanks

zhenyzha commented:
```
 (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.arm64-pci: STARTED
 (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.arm64-pci: ERROR: Failures occurred while postprocess:\n\nHost dmesg verification failed: Found unexpected failures in host dmesg log Please check host dmesg log /root/avocado/job-results/job-2022-08-22T22.44-c20fbcb/test-results/1-Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio... (107.36 s)
 (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_raw_format.arm64-pci: STARTED
 (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_raw_format.arm64-pci: PASS (95.95 s)
 (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_luks_format.arm64-pci: STARTED
 (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_luks_format.arm64-pci: PASS (100.56 s)
```

The error has nothing to do with the patch
The commit limit is on s390x, but this is a platform-wide change, so the commit description needs to be updated. other acks for me

qingwangrh commented on generic/tests/cfg/nfs_corrupt.cfg:
```diff
@@ -16,9 +16,7 @@
-    s390x:
-        bootindex_image1 = 0
-        bootindex_stg = 1
+    blk_extra_params_stg = "serial=TARGET_DISK"
```
suggest change to like :
serial=TARGET_DISK
blk_extra_params_stg += "serial=${serial}"

fbq815 replied:
updated

qingwangrh commented on generic/tests/nfs_corrupt.py:
```diff
@@ -215,7 +216,12 @@ def check_vm_status(vm, status):
     session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
 
-    nfs_devname = get_nfs_devname(params, session)
+    if stg_params.get('blk_extra_params'):
```
nfs_devname = utils_misc.get_linux_drive_path(session,stg_params["serial"])

fbq815 replied:
updated

qingwangrh commented on generic/tests/nfs_corrupt.py:
```diff
@@ -215,7 +216,12 @@ def check_vm_status(vm, status):
 
-    nfs_devname = get_nfs_devname(params, session)
+    if stg_params.get('blk_extra_params'):
+        pattern = re.search(r"(serial|wwn)=(\w+)",
```
both removable, just remain  nfs_devname = utils_misc.get_linux_drive_path(session,
                                                          stg_params["serial"])

fbq815 commented:
> ```
>  (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.arm64-pci: STARTED
>  (1/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.arm64-pci: ERROR: Failures occurred while postprocess:\n\nHost dmesg verification failed: Found unexpected failures in host dmesg log Please check host dmesg log /root/avocado/job-results/job-2022-08-22T22.44-c20fbcb/test-results/1-Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio... (107.36 s)
>  (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_raw_format.arm64-pci: STARTED
>  (2/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_raw_format.arm64-pci: PASS (95.95 s)
>  (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_luks_format.arm64-pci: STARTED
>  (3/3) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.aarch64.io-github-autotest-qemu.nfs_corrupt.with_luks_format.arm64-pci: PASS (100.56 s)
> ```
> 
> The error has nothing to do with the patch The commit limit is on s390x, but this is a platform-wide change, so the commit description needs to be updated. other acks for me

updated, and as the comment said what I deleted is s390x only

fbq815 commented:
Test result:
JOB ID     : c60a8f8e326d89412b337110b904442a177c6689
JOB LOG    : /root/avocado/job-results/job-2022-08-23T03.56-c60a8f8/job.log
 (1/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (1/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (60.16 s)
 (2/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (2/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (57.53 s)
 (3/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (3/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (63.23 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-08-23T03.56-c60a8f8/results.html
JOB TIME   : 184.02 s

maxujun commented:
(1/3) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format: PASS (247.44 s)
 (2/3) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.nfs_corrupt.with_raw_format: PASS (239.30 s)
 (3/3) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.nfs_corrupt.with_luks_format: PASS (237.60 s)

PaulYuuu commented on generic/tests/cfg/nfs_corrupt.cfg:
```diff
@@ -16,9 +16,8 @@
-    s390x:
-        bootindex_image1 = 0
-        bootindex_stg = 1
+    serial = TARGET_DISK
```
serial is a common option for some devices, better to add a prefix.

qingwangrh replied:
agree

fbq815 replied:
updated

PaulYuuu commented on generic/tests/nfs_corrupt.py:
```diff
@@ -215,7 +216,12 @@ def check_vm_status(vm, status):
+    if stg_params.get('blk_extra_params'):
+        pattern = re.search(r"(serial|wwn)=(\w+)",
+                            stg_params["blk_extra_params"], re.M)
+        if pattern is not None:
```
With this change, no need to check the pattern. The only concern is that if users re-define the blk_extra_params, then we can not make sure serial=xxx exists. But if no nfs_devname, this case will fail.

qingwangrh replied:
We can consider it is pre-condition.  otherwise if serial non-existent, the other keys also may non-existent.  They are not the test points.
If the expected key does not exist result in failed result, the coder should check its code or configuration.   

Anyway the both are accepable to me.

fbq815 replied:
updated, please check if the new update could fix the issue

PaulYuuu commented on generic/tests/cfg/nfs_corrupt.cfg:
```diff
@@ -16,9 +16,8 @@
-        bootindex_image1 = 0
-        bootindex_stg = 1
+    serial = TARGET_DISK
+    blk_extra_params_stg += "serial=${serial}"
```
For `+=` operation, you should use `,serial=${serial}`, an example is `blk_extra_params_stg = key=value` + `blk_extra_params_stg += "serial=${serial}"`, it will become `blk_extra_params_stg = key=valueserial=${serial}"`. If qcontainer cannot handle it, it's bug of qcontainer, or you have to use `=` operation here.

qingwangrh replied:
agree

fbq815 replied:
updated

fbq815 commented:
JOB ID     : ada1f27c34a5c4f45eee68281c018758d6875a52
JOB LOG    : /root/avocado/job-results/job-2022-08-24T05.54-ada1f27/job.log
 (1/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (1/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (63.76 s)
 (2/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (2/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (58.18 s)
 (3/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (3/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (63.55 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-08-24T05.54-ada1f27/results.html
JOB TIME   : 188.62 s

qingwangrh commented:
LGTM

fbq815 commented:
Test result:
JOB ID     : 57ebbd5d18b57e9390eb33c53eef0d05477487b9
JOB LOG    : /root/avocado/job-results/job-2022-08-25T04.32-57ebbd5/job.log
 (1/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: STARTED
 (1/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_qcow2_format.s390-virtio: PASS (60.33 s)
 (2/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: STARTED
 (2/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_raw_format.s390-virtio: PASS (57.67 s)
 (3/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: STARTED
 (3/3) Host_RHEL.m9.u1.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.nfs_corrupt.with_luks_format.s390-virtio: PASS (63.01 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-08-25T04.32-57ebbd5/results.html
JOB TIME   : 184.26 s
@PaulYuuu please help me merge this patch
